### PR TITLE
Configurable nuget sources

### PR DESCRIPTION
**Pull request by skolima**
The `dotnet` version now also does a restore first, because `dotnet add` only accepts one target repository name and would fail if dependencies for the update are missing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,7 +92,7 @@ It is similar to [this rule of preferring to use a parameter of a base type or i
 
 ## Limitations
 
-You will need command line versions of `git` and `dotnet` installed.
+You will need command line version of `dotnet` installed.
 
 It currently only runs on windows due to using `cmd` to invoke command-line processes for `git` and `dotnet`.
```
Comment (AnthonySteele) [inline README.md]: Also remove mention of git here.

**Comment by AnthonySteele** [comment]
👍 Looks ok and we need this feature, many orgs will have internal nuget feeds serving packages that need auto-updating.

File: NuKeeper/NuGet/Api/PackageSearchMedatadataWithSource.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        public string Summary => _originalMetadata.Summary;
+
+        public string Tags => _originalMetadata.Tags;
+
+        public string Title => _originalMetadata.Title;
+
+        public bool IsListed => _originalMetadata.IsListed;
+    }
+}
```
Comment (AnthonySteele) [inline NuKeeper/NuGet/Api/PackageSearchMedatadataWithSource.cs]: Do we need *all* of these properties surfaced? I would leave most of them off until if/when we need them.

Comment (skolima) [inline NuKeeper/NuGet/Api/PackageSearchMedatadataWithSource.cs]: TBH, I've implemented the full NuGet interface here, as it made for a much easier refactoring - and the implementation is autogenerated by R#. If you want, I can remove the interface and just trim it to what we actually use.

Comment (AnthonySteele) [inline NuKeeper/NuGet/Api/PackageSearchMedatadataWithSource.cs]: I would remove the interface and trim it down, yes.

**Pull request merged**